### PR TITLE
Update Settings Tab Icons [Fiji] #1659

### PR DESCRIPTION
**Pull request by nimitparekh2020**
- implementation of setting screen with live data integration
@hboon - Currently I implemented design as per zeplin guideline mention into #1659 So review the code and give me feedback as per that.

### CODE REVIEW

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        }
+    }
+    let containerView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.clipsToBounds = true // this will make sure its children do not go out of the boundary
+        return view
+    }()
+    
+    let settingIconImage: UIImageView = {
```
Comment (hboon) [inline AlphaWallet/Settings/Views/SettingTableViewCell.swift:53]: Most of these properties should be private

File: AlphaWallet/Settings/ViewModels/SettingModel.swift [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+//  settingModel.swift
+//  AlphaWallet
+//
+//  Created by Nimit Parekh on 06/04/20.
+//
+
+import UIKit
+
+struct SettingModel {
+    let title: String?
```
Comment (hboon) [inline AlphaWallet/Settings/ViewModels/SettingModel.swift:11]: Are any of these properties ever nil? If not, make them non-optional

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+           let label = UILabel()
+            label.font = R.font.sourceSansProRegular(size: 17)
+           label.textColor = R.color.black()
+           label.translatesAutoresizingMaskIntoConstraints = false
+           return label
+       }()
+       
+       let settingSubTitle: UILabel = {
+           let label = UILabel()
+           label.font = R.font.sourceSansProRegular(size: 12)
```
Comment (hboon) [inline AlphaWallet/Settings/Views/SettingTableViewCell.swift:71]: Use `Fonts.regular(size: 12)`

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        didSet {
+            guard let settingItem = settings else { return }
+            if let title = settingItem.title {
+                settingTitle.text = title
+            }
+            if let subtitle = settingItem.subTitle {
+                settingSubTitle.text = subtitle
+                let subTitleLength = subtitle.count
+                if subTitleLength > 0 {
+                    titleTopConstraint = settingTitle.centerYAnchor.constraint(equalTo: self.containerView.centerYAnchor, constant: -10)
```
Comment (hboon) [inline AlphaWallet/Settings/Views/SettingTableViewCell.swift:37]: This is very harder to maintain. Can you make 2 different cell classes? 1 with subtitle, 1 without.

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+            if section == 3 {
+                return 0
+            }
+            return 50
+        }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: "SettingHeaderView") as! SettingViewHeader
+        if section == 0 {
+            headerView.title = "WALLET"
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:123]: Localize.

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+    
+        func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+            if section == 3 {
+                return 0
+            }
+            return 50
+        }
+
+    func tableView(_ tableView: UITableView, viewForHeaderInSection section: Int) -> UIView? {
+        let headerView = tableView.dequeueReusableHeaderFooterView(withIdentifier: "SettingHeaderView") as! SettingViewHeader
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:121]: Extract magic string

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+        } else {
+            headerView.title = "HELP"
+        }
+        return headerView
+    }
+}
+extension SettingsViewController2: SettingTableViewCellDelegate {
+    func onOffPassCode(cell: SettingTableViewCell, switchState isOnOff: Bool) {
+        if isOnOff {
+            print("Switch is On")
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:135]: Remove debug prints (but I understand this is still WIP)

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+        } else if section == 1 {
+            headerView.title = "SYSTEM"
+        } else {
+            headerView.title = "HELP"
+        }
+        return headerView
+    }
+}
+extension SettingsViewController2: SettingTableViewCellDelegate {
+    func onOffPassCode(cell: SettingTableViewCell, switchState isOnOff: Bool) {
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:133]: `isOnOff: Bool`

You probably meant `isOn: Bool` instead.

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+                case 2:
+                    cell.settings = settingsHelpData[indexPath.row]
+                default:
+                   cell.settings = settingsHelpData[indexPath.row]
+                }
+                return cell
+            }
+        }
+        
+        func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:104]: This should be taken care by Auto Layout

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+            if indexPath.row == 1 && indexPath.section == 0 {
+                return 80
+            } else if indexPath.section == 3 {
+                return  50
+            }
+            return 60
+        }
+    
+        func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:113]: This should be taken care by Auto Layout

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+        
+        settingTableView.dataSource = self
+        settingTableView.delegate = self
+        
+        settingTableView.register(SettingViewHeader.self, forHeaderFooterViewReuseIdentifier: "SettingHeaderView")
+        settingTableView.register(SettingTableViewCell.self, forCellReuseIdentifier: "settingCell")
+        settingTableView.register(AppInfoTableCell.self, forCellReuseIdentifier: "appInfoCell")
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
```
Comment (hboon) [inline AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:49]: Doesn't look like this is used

**Comment by hboon** [comment]
@nimitparekh2020 we use the "WIP. Don't Merge" label/tag, adding/removing them actively to indicate if a PR is good for merging or not.

**Comment by nimitparekh2020** [comment]
Okay, I'll follow all the comment and work on it and update you.

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+
+import UIKit
+
+protocol SettingTableViewCellDelegate: class {
+    func onOffPassCode(cell: SettingTableViewCell, switchState isOnOff: Bool)
+}
+
+class SettingTableViewCell: UITableViewCell {
+
+    override func awakeFromNib() {
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/Views/SettingTableViewCell.swift:16]: Removed.

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift [hunk 13]
```diff
@@ -0,0 +1,129 @@
...
+        super.awakeFromNib()
+        // Initialization code
+    }
+
+    override func setSelected(_ selected: Bool, animated: Bool) {
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/Views/SettingTableViewCell.swift:21]: Removed.

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+            }
+            if let icon = settingItem.icon {
+                settingIconImage.image = icon
+            }
+        }
+    }
+    let containerView: UIView = {
+        let view = UIView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.clipsToBounds = true // this will make sure its children do not go out of the boundary
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/Views/SettingTableViewCell.swift:49]: Removed.

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift (hunk 3 above)
Comment (nimitparekh2020) [reply AlphaWallet/Settings/Views/SettingTableViewCell.swift:71]: In Zeplin these fonts used.

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 15]
```diff
@@ -0,0 +1,140 @@
...
+            case 2:
+                return settingsHelpData.count
+            default:
+                return settingsFooterData.count
+            }
+        }
+        
+        func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+            if indexPath.section == 3 {
+                let cells = tableView.dequeueReusableCell(withIdentifier: "appInfoCell", for: indexPath) as? AppInfoTableCell
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:77]: Done

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+extension SettingsViewController2: UITableViewDataSource, UITableViewDelegate {
+        public func numberOfSections(in tableView: UITableView) -> Int {
+            return 4
+        }
+    
+        func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+            switch section {
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:63]: Done

File: AlphaWallet/Settings/ViewModels/SettingModel.swift [hunk 17]
```diff
@@ -0,0 +1,18 @@
+//
+//  settingModel.swift
+//  AlphaWallet
+//
+//  Created by Nimit Parekh on 06/04/20.
+//
+
+import UIKit
+
+struct SettingModel {
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewModels/SettingModel.swift:10]: Done

File: AlphaWallet/Settings/ViewModels/SettingModel.swift [hunk 18]
```diff
@@ -0,0 +1,18 @@
...
+    let title: String?
+    let subTitle: String?
+    let icon: UIImage?
+}
+struct SettingFooterModel {
```
[the remark this answers is not in the document]
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewModels/SettingModel.swift:15]: Done

File: AlphaWallet/Settings/Views/SettingTableViewCell.swift (hunk 1 above)
Comment (nimitparekh2020) [reply AlphaWallet/Settings/Views/SettingTableViewCell.swift:53]: Done

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift (hunk 6 above)
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:121]: Done

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift (hunk 8 above)
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:133]: Done

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift (hunk 11 above)
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:49]: Done

File: AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift (hunk 7 above)
Comment (nimitparekh2020) [reply AlphaWallet/Settings/ViewControllers/SettingsViewController2.swift:135]: Done

**Pull request closed without merging**